### PR TITLE
E2e appium request token3

### PR DESCRIPTION
**Pull request by Andepande**
**Description**

- E2e test for Request token flow

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: wdio/screen-objects/RequestTokenScreen.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  async closePaymentRequest() {
+    await Gestures.tap(this.requestCloseButton);
+  }
+
+  async inputSearchRequestField(searchReq) {
+    await Gestures.typeText(this.requestSearchInput, searchReq);
+  }
+
+  async tapBackButtonOnSearch(searchReq) {
```
Comment (cortisiko) [inline wdio/screen-objects/RequestTokenScreen.js:57]: the variable `searchReq` is not used. probably remove it as a parameter.

File: wdio/screen-objects/RequestTokenScreen.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  async isTextElementNotDisplayed(text) {
+    await expect(Selectors.getXpathElementByText(text)).not.toBeDisplayed();
+  }
+
+  async closePaymentRequest() {
+    await Gestures.tap(this.requestCloseButton);
+  }
+
+  async inputSearchRequestField(searchReq) {
```
Comment (cortisiko) [inline wdio/screen-objects/RequestTokenScreen.js:53]: mind spelling out this variable.

File: wdio/step-definitions/request-token.steps.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+When(/^I type "([^"]*)?" into the Request Amount field/, async (amount) => {
+  await RequestTokenScreen.typeAmountInRequest(amount);
+});
+
+Then(
+  /^I type "([^"]*)?" in the Search Assets field/,
+  async (seaarchRequest) => {
+    const timeout = 1000;
+    await driver.pause(timeout);
+    await RequestTokenScreen.inputSearchRequestField(seaarchRequest);
```
Comment (cortisiko) [inline wdio/step-definitions/request-token.steps.js:20]: typo in `seaarchRequest`

File: wdio/screen-objects/testIDs/Screens/RecieveToken.testIds.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export const REQUEST_AMOUNT_INPUT = 'request-amount-input';
```
Comment (cortisiko) [inline wdio/screen-objects/testIDs/Screens/RecieveToken.testIds.js:1]: Can you rename this file to `RequestToken.testIDs.js`

File: wdio/screen-objects/RequestTokenScreen.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  REQUEST_SEARCH_SCREEN,
+  PAYMENT_REQUEST_QR_CODE_CLOSE_ICON,
+} from '../screen-objects/testIDs/Screens/RecieveToken.testIds';
+
+class RequestTokenScreen {
+  get requestAmount() {
+    return Selectors.getElementByPlatform(REQUEST_AMOUNT_INPUT);
+  }
+
+  get requestCloseButton() {
```
Comment (cortisiko) [inline wdio/screen-objects/RequestTokenScreen.js:17]: Create another Screen class for the SendLink screen. The `requestCloseButton`, `closeRequestPaymentQRIcon` && `closePaymentRequest` should be in the SendLink class.

**Review by cortisiko — CHANGES_REQUESTED** [review]
left a few comments @Andepande !

**Review by cortisiko — APPROVED** [review]
Looks good to me. Before you merge:
- Ensure detox is not broken. In particular, the `request-token-flow.spec.js`
-  Get one more dev review

**Review by SamuelSalas — APPROVED** [review]

**Pull request merged**